manifests/plugin.pp

```diff
@@ -27,10 +29,20 @@
       }
     }
 
-    concat::fragment { $title:
-      target  => $path,
-      content =>  template('hindsight/plugin.cfg.erb'),
-      order   => $order,
+    # if one gives a CONTENT or SOURCE arg, take that
+    if $content != undef or $source != undef {
```

Comment:
This leads to unnecessary code repetition, it would be cleaner to use something like:
```puppet
 if $content {
      Concat::Fragment<| title == $title |> {
        content => $content,
      }
} elsif $source  {
      Concat::Fragment<| title == $title |> {
        source => $source,
      }
} else {
      Concat::Fragment<| title == $title |> {
        content =>  template('hindsight/plugin.cfg.erb'),
      }
}
```
The `Concat::Fragment<| title == $title |>` statement is extending previously declared `concat::fragment { $title: }` statement.

Reply:
I don't really agree with your point, but I'll fix it.

If you want tests for this, I'd like your help. I tried making one, but the function "template()" is absent, neither inline_template().

It seems the `` if $manage_service then notify[] `` is currently done twice. I could fix that in an extra commit too?

Reply:
Multiple notifies should be fine, the action is executed just once. I'll try to have a look at the tests.